components/teams-service/storage/postgres/postgres.go

```diff
@@ -258,42 +258,56 @@ func (p *postgres) GetTeams(ctx context.Context) ([]storage.Team, error) {
 // server code before we get here if the team in question was filtered, but if we call this
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
-func (p *postgres) RemoveUsers(ctx context.Context, id string, userIDs []string) (storage.Team, error) {
+func (p *postgres) RemoveUsers(ctx context.Context, id string, userIDs []string) (updatedUserIDs []string, err error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
 	tx, err := p.db.BeginTx(ctx, nil)
 	if err != nil {
-		return storage.Team{}, p.processError(err)
+		return nil, p.processError(err)
 	}
 
-	res, err := tx.ExecContext(ctx,
-		`DELETE FROM teams_users_associations
-		WHERE user_id = ANY($1) AND team_db_id = team_db_id($2)`,
-		pq.Array(userIDs), id)
+	projectsFilter, err := ProjectsListFromContext(ctx)
 	if err != nil {
-		return storage.Team{}, p.processError(err)
+		return nil, p.processError(err)
 	}
-	num, err := res.RowsAffected()
+
+	var dbID int
+	var teamProjects []string
+	// ensure the team exists and isn't filtered out by the project filter
+	err = tx.QueryRowContext(ctx,
+		`SELECT db_id, projects FROM teams
+		WHERE id = $1 AND projects_match(projects, $2::TEXT[])`,
+		id, pq.Array(projectsFilter)).
+		Scan(&dbID, pq.Array(&teamProjects))
 	if err != nil {
-		return storage.Team{}, p.processError(err)
+		return nil, p.processError(err)
 	}
 
-	var team storage.Team
-	if num == 0 { // no changes
-		team, err = p.getTeam(ctx, tx, id)
-	} else {
-		team, err = p.touchTeam(ctx, tx, id)
-	}
+	_, err = tx.ExecContext(ctx,
+		`DELETE FROM teams_users_associations
+		WHERE user_id = ANY($1) AND team_db_id=$2`,
+		pq.Array(userIDs), dbID)
 	if err != nil {
-		return storage.Team{}, p.processError(err)
+		return []string{}, p.processError(err)
 	}
 
 	err = tx.Commit()
 	if err != nil {
-		return storage.Team{}, p.processError(err)
+		return nil, p.processError(err)
 	}
-	return team, nil
+
+	// now that the transaction is complete and the team's users are updated,
```

Comment:
Why do we do this _after_ the transaction? If we ran it inside, would the response differ?

Reply:
It was failing when I originally had it inside the transaction so I moved it out. But I just moved it back in and it still passes, so I must have misunderstood the original test failure.